tests/helpers/assertions.py

```diff
@@ -395,6 +395,33 @@ def has_dim_prop(fake_services, dim_name, dim_value, prop_name, prop_value=None)
         return props.get(prop_name) == prop_value
     return False
 
+def has_dim_set_prop(fake_services, dim_name, dim_value, prop_name, prop_values=[]):
+    """
+    Tests if the given dimension has a set property containing all of the given values. If prop_values is None,
+    tests for the presence of the property regardless of value. Only tests if the given properties are present, does
+    not test that they are the only values in the given property set
+    """
+    dim = fake_services.dims[dim_name].get(dim_value, {})
+    props = dim.get(prop_name, [])
+    has_props = False
+    if props is not None:
+        has_props = True
+        for v in prop_values:
+            has_props = True and v in props
+    return has_props
+
+def not_has_dim_set_prop(fake_services, dim_name, dim_value, prop_name, prop_values):
```

Comment:
Name should be `has_no_dim_prop`.

Reply:
Also it should be able to handle a scalar value for `prop_values`, as well as a list.

Reply:
And then call it `prop_value`.

Reply:
Updated to use `*prop_value` as the last parameter, but that also means that we can't use named arguments when this gets used in `wait_for`

Reply:
I would make it `prop_value: Union[str, List[str]]` (you don't have to include Py3 typing annotations, but they are nice to have) and then have `if isinstance(prop_value, str): prop_value = [prop_value]` or something like that to make it work nicely with partials.